app/src/main/java/com/kenchen/capstonenewsapp/views/news/NewsViewModel.kt

```diff
@@ -0,0 +1,52 @@
+package com.kenchen.capstonenewsapp.views.news
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.kenchen.capstonenewsapp.App.Companion.remoteApi
+import com.kenchen.capstonenewsapp.model.Article
+import com.kenchen.capstonenewsapp.networking.RemoteResult
+import com.kenchen.capstonenewsapp.utils.mapException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class NewsViewModel : ViewModel() {
+
+    private val _headLineNewsLiveData = MutableLiveData<RemoteResult<List<Article>>>()
+    val headLineNewsLiveData: LiveData<RemoteResult<List<Article>>> = _headLineNewsLiveData
+
+    private val test = MutableLiveData<List<String>>()
+
+    private val _newsLoadingStateLiveData = MutableLiveData<NewsLoadingState>()
+    val newsLoadingStateLiveData: LiveData<NewsLoadingState> = _newsLoadingStateLiveData
+
+    fun onActivityReady() {
+        // prevent refetching data when orientation changes (save resource)
+        if (_headLineNewsLiveData.value == null) {
+            getTopHeadlinesNewsByCountry()
+        }
```

Comment:
Nice work here considering the orientation change! There's still one loop hole here, where if the initial API call takes a long time (say, the user has poor internet connectivity) and then the user rotates the screen, the `_headLinesNewsLiveData.value` will be null and a subsequent coroutine will run to fetch the articles from the API.

What you could do instead is have a `Job` variable:

```kotlin
private var job: Job? = null
```

And then set it in your `getTopHeadlinesNewsByCountry()` method along with a check to see if there's a prev job running and active:

```kotlin
private fun getTopHeadlinesNewsByCountry() {
    _newsLoadingStateLiveData.value = NewsLoadingState.LOADING
    if (job?.isActive == true) return
    
    job = viewModelScope.launch(Dispatchers.IO) {
```